lib/freedom_patches/net_http_header.rb

```diff
@@ -0,0 +1,17 @@
+# frozen_string_literal: true
+
+module NetHTTPHeaderPatch
+  def initialize_http_header(initheader)
+    # If no user-agent is set, set it to the default
+    initheader ||= {}
+    user_agent_key =
+      initheader.keys.find { |key| key.to_s.downcase == "user-agent" } || "User-Agent".to_sym
+    initheader[
+      user_agent_key
+    ] ||= "Discourse/#{Discourse::VERSION::STRING}-#{Discourse.git_version}; +https://www.discourse.org/"
```

Comment:
Minor nit, `"Discourse/#{Discourse::VERSION::STRING}-#{Discourse.git_version}; +https://www.discourse.org/"` can probably be moved into a constant that can be referenced from tests without having to duplicate the logic in the string.

Reply:
Hehe had the same comment about moving it to `lib/discourse.rb` so we can do `Discourse.user_agent`